back/boxtribute_server/models/crud.py

```diff
@@ -435,3 +438,48 @@ def create_qr_code(box_label_identifier=None):
             box.save()
 
     return new_qr_code
+
+
+def get_box_history(box_id):
+    entries = []
+    for raw_entry in DbChangeHistory.select().where(
+        DbChangeHistory.table_name == "stock",
+        DbChangeHistory.record_id == box_id,
+    ):
+        changes = raw_entry.changes
+        if changes == "items":
+            changes = (
+                f"changed the number of items from {raw_entry.from_int} to "
+                + f"{raw_entry.to_int};"
+            )
+
+        elif changes == "product_id":
+            old_product = Product.get_by_id(raw_entry.from_int)
+            new_product = Product.get_by_id(raw_entry.to_int)
+            changes = (
+                f"changed product type from {old_product.name} to "
+                + f"{new_product.name};"
+            )
+
+        elif changes == "location_id":
+            old_location = Location.get_by_id(raw_entry.from_int)
+            new_location = Location.get_by_id(raw_entry.to_int)
+            changes = (
+                f"changed box location from {old_location.name} to "
+                + f"{new_location.name};"
+            )
+
+        elif changes == "size_id":
+            old_size = Size.get_by_id(raw_entry.from_int)
+            new_size = Size.get_by_id(raw_entry.to_int)
+            changes = f"changed size from {old_size.label} to {new_size.label};"
+
+        elif changes.startswith("comments"):
+            changes = changes.replace("comments changed", "changed comments")
+
+        elif changes.startswith("Record"):
+            changes = changes.replace("Record created", "created record")
+
```

Comment:
@pylipp, you can also make a history recode for changes to the state of the box. This is also part of this [PR](https://github.com/boxwise/dropapp/pull/502) in the DropApp. This can, however, be done as a follow-up PR once the mentioned PR has been merged.

Reply:
I see! I don't even take into account that changing the state is logged in `save_update_to_history`. The relevant bit in the dropapp PR seems to be [this](https://github.com/boxwise/dropapp/pull/502/files#diff-d3e74dca303d72c858762488d8e0381e3382a6b2dddad344931aeb342317f716R91-R94), though I don't fully understand the difference of the two messages (why is the information text-only and not using `from_int` and `to_int` fields of the history table?)